src/bin/firebase.js

```diff
@@ -37,28 +38,34 @@ var utils = require("../utils");
 var cmd;
 
 var logFilename = path.join(process.cwd(), "/firebase-debug.log");
-logger
-  .add(winston.transports.Console, {
-    level: process.env.DEBUG ? "debug" : "info",
-    showLevel: false,
-    colorize: true,
-  })
-  .add(winston.transports.File, {
-    level: "debug",
-    filename: logFilename,
-    json: false,
-    formatter: function(input) {
-      input.message = ansiStrip(input.message);
-      return ["[" + input.level + "]", input.message].join(" ");
-    },
-  });
 
-var debugging = false;
-if (_.includes(args, "--debug")) {
-  logger.transports.console.level = "debug";
-  debugging = true;
+if (!process.env.DEBUG && _.includes(args, "--debug")) {
+  process.env.DEBUG = true;
+}
+
+function tryStringify(value) {
+  if (typeof value === "string") return value;
+
+  try {
+    return JSON.stringify(value);
+  } catch {
+    return value;
+  }
 }
 
+logger.add(
+  new winston.transports.File({
+    level: "debug",
+    filename: logFilename,
+    format: winston.format.printf(
+      (info) =>
+        `[${info.level}] ${ansiStrip(
+          [info.message, ...(info[SPLAT] || [])].map(tryStringify).join(" ")
+        )}`
```

Comment:
This still isn't producing the correct output.

![image](https://user-images.githubusercontent.com/160236/75733908-5c8b0880-5cab-11ea-9ce0-1f45e2c82cef.png)

Reply:
Erm... I'm getting `[debug] [2020-03-06T19:12:56.114Z] <<< HTTP RESPONSE 200 {"x-google-esf-cloud-client-params":"backend_servi...` this with a `deploy` right?